packages/atomic/src/components/common/interface/store.ts

```diff
@@ -51,11 +51,15 @@ export function createAtomicCommonStore<
       data: StoreData[T][U] & {facetId: U; element: HTMLElement}
     ) {
       if (stencilStore.state[facetType][data.facetId]) {
-        return;
+        stencilStore.state.facetElements =
+          stencilStore.state.facetElements.filter(
+            (facetElement) =>
+              facetElement.getAttribute('facet-id') !== data.facetId
+          );
```

Comment:
I think we should extract that into a separate function like `removeFacetElements(facetId)`.